web/src/actions/auth.js

```diff
@@ -101,22 +95,39 @@ export const login = (username, password) => dispatch => {
   dispatch(requestLogin());
   authenticateUser(username, password)
     .then(async res => {
+      if (res.status === 'LOCKED_OUT') {
+        return dispatch(failLoginLocked());
+      }
+      
       if (res.status === 'MFA_REQUIRED') {
-        return retrieveMFA(res).then(() => {
-          dispatch(completeFirstStage());
-        });
+        const mfaFactor = res.factors.find(
+          factor => factor.provider === 'OKTA'
+        );
+        // Currently we are only supporting one additional factor
+        const mfaType = res.factors[0].factorType;
```

Comment:
@radavis @thetif any concerns with this line?

Reply:
line 103 is getting the first factor associated with OKTA, so you can just say
`const { factorType: mfaType } = mfaFactor;`

also you should put that part after line 109 so that it won't try to destructure if it is undefined

Reply:
@tbolt can you change this just in case their first factor isn't from OKTA?